src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -738,6 +738,59 @@ def makeTransferRucio(self, wflow, dataIn, subLevel, blocks, dataSize, nodes, no
             self.logger.info(msg, wflow.getName(), dids, rseExpr, ruleAttrs)
         return success, transferId
 
+    def sendAlert(self, alertName, severity, summary, description, service, endSecs = 1 * 60 * 60):
+        """
+        Send alert to Prometheus, wrap function in a try-except clause
+        """
+        try:
+            # alert to expiry in an hour from now
+            self.alertManagerApi.sendAlert(alertName, severity, summary, description,
+                                           service, endSecs)
+        except Exception as ex:
+            self.logger.exception("Failed to send alert to %s. Error: %s", self.alertManagerUrl, str(ex))
+
+    def alertPUMisconfig(self, workflowName):
+        """
+        Send alert to Prometheus with PU misconfiguration error
+        """
+        alertName = "{}: PU misconfiguration error. Workflow: {}".format(self.alertServiceName,
+                                                                         workflowName)
+        alertSeverity = "high"
+        alertSummary = "[MSTransferor] Workflow cannot proceed due to some PU misconfiguration."
+        alertDescription = "Workflow: {} could not proceed due to some PU misconfiguration,".format(workflowName)
+        alertDescription += "so it will be skipped."
+        self.sendAlert(alertName, alertSeverity, alertSummary, alertDescription,
+                       self.alertServiceName)
+        self.logger.critical(alertDescription)
+
+    def alertUnknownTransferError(self, workflowName):
+        """
+        Send alert to Prometheus with unknown transfer error
+        """
+        alertName = "{}: Transfer request error. Workflow: {}".format(self.alertServiceName,
+                                                                         workflowName)
+        alertSeverity = "high"
+        alertSummary = "[MSTransferor] Unknown exception while making Transfer Request."
```

Comment:
I think we do not need the initial capital letters here. It could be "transfer request", no? Same comment for the line below.

Kenyi, in addition to that, this log exception is already executed on the caller function. Perhaps we can leave it only there and remove it from this method?